io/plugins/eu.esdihumboldt.hale.io.shp.test/src/eu/esdihumboldt/hale/io/shp/ShapefileInstanceWriterTest.groovy

```diff
@@ -1443,4 +1456,23 @@ class ShapefileInstanceWriterTest {
 			assertEquals(2, num)
 		}
 	}
+
+	@Test
+	void testCPGFileCreation() {
```

Comment:
This test, tests the functionality of writing just a standalone file. Would it be possible to create another test where you can confirm creation of `cpg` file in the complete process?
You can refer to [this test case](https://github.com/halestudio/hale/blob/a970f6e19cac38578fb00eb75baa31808df22423/io/plugins/eu.esdihumboldt.hale.io.shp.test/src/eu/esdihumboldt/hale/io/shp/ShapefileInstanceWriterTest.groovy#L208) and may be replicate it and when you have [reference to file](https://github.com/halestudio/hale/blob/a970f6e19cac38578fb00eb75baa31808df22423/io/plugins/eu.esdihumboldt.hale.io.shp.test/src/eu/esdihumboldt/hale/io/shp/ShapefileInstanceWriterTest.groovy#L232C40-L232C40) probably you can there also assert for the creation of `cpg` file.

Reply:
Sure!

Reply:
> But I think that this test is already done in one of the already created test, example simpleSchemaWriteTest(), This test would create the .shp and .cpg files.

Yes, I think it would also be fine to extend existing tests to also check if the CPG file was written and the content is correct.

Reply:
A bonus would be adding a test where a Shapefile with special characters is written and read again, with a specific charset configured for both, to verify that charset settings for writer and reader work as expected.

Reply:
Yes, it must be. But still it would be good to verify either in the same test or in an additional test.